app/src/main/java/org/dhis2/uicomponents/map/managers/TeiMapManager.kt

```diff
@@ -243,6 +281,42 @@ class TeiMapManager(mapView: MapView) : MapManager(mapView) {
         return featureToReturn
     }
 
+    override fun findFeatures(
+        source: String,
+        propertyName: String,
+        propertyValue: String
+    ): List<Feature>? {
```

Comment:
I'm worry about managers become a box full of find features, so far we have 3 in TeiMapManager? What do you think?  is it worthy to have a meeting to do some brainstorming? 

Also, not sure if the manager should be responsible for finding all features

Reply:
I have already located where these findFeature methods are used so i would like to create a class in charge of this.
Once we have this class the testing will be quite straightforward

Reply:
:+1: I did not mean to refactor everything, but if it was a way to actually not add another findFeature. I guess it depens on carousel view?.

Reply:
Mostly yes, it also has some dependency with the onMapClickListener, but I think moving all these findFeature to the carousel would be the best